src/components/QuickOpenModal.js

```diff
@@ -356,6 +356,24 @@ export class QuickOpenModal extends Component<Props, State> {
     });
   };
 
+  shouldShowErrorEmoji() {
+    const { query } = this.props;
+
+    if (this.isGotoQuery()) {
+      return !/^:\d*$/.test(query);
+    }
+
+    if (
+      this.isVariableQuery() ||
+      this.isFunctionQuery() ||
+      this.isShortcutQuery()
+    ) {
+      return !this.getResultCount();
```

Comment:
It seems that going by this logic we could completely eliminate any checks in `shouldShowErrorEmoji()` in `SearchInput.js` in favor of just passing a true/false based on the checks in the corresponding dedicated module.